src/reset-password/ResetPasswordPage.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Button, Input, ValidationFormGroup } from '@edx/paragon';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import messages from './messages';
+import { resetPassword, validateToken } from './data/actions';
+import { resetPasswordResultSelector } from './data/selectors';
+import InvalidTokenMessage from './InvalidToken';
+import ResetSuccessMessage from './ResetSuccess';
+import Spinner from './Spinner';
+
+
+const ResetPasswordPage = (props) => {
+  const { intl } = props;
+
+  const [newPasswordInput, setNewPasswordValue] = useState('');
+  const [confirmPasswordInput, setConfirmPasswordValue] = useState('');
+  const [passwordValid, setPasswordValidValue] = useState(true);
+  const [passwordMatch, setPasswordMatchValue] = useState(true);
+
+  const handleNewPasswordChange = (e) => {
+    const newPassword = e.target.value;
+    setNewPasswordValue(newPassword);
+    const isValid = newPassword.length >= 8 && newPassword.match(/\d+/g);
+    setPasswordValidValue(isValid !== false);
+  };
+  const handleConfirmPasswordChange = (e) => {
+    const confirmPassword = e.target.value;
+    setConfirmPasswordValue(confirmPassword);
+    setPasswordMatchValue(confirmPassword === newPasswordInput);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (newPasswordInput === '') {
+      setPasswordValidValue(false);
+    }
+    if (passwordValid && passwordMatch) {
+      const formPayload = {
+        new_password1: newPasswordInput,
+        new_password2: confirmPasswordInput,
+      };
+      props.resetPassword(formPayload, props.token);
+    }
+  };
+
+  if (props.token_status === 'pending') {
+    const { token } = props.match.params;
+    if (token) {
+      props.validateToken(token);
+      return <Spinner text='Token validation in progress .. ' />;
+    }
+  } else if (props.token_status === 'invalid') {
+    return (<InvalidTokenMessage />);
+  } else {
+    if (props.status === 'success'){
+      return (<ResetSuccessMessage />);
+    } else{
+      return (
+        <>
+          <div className="d-flex justify-content-center reset-password-container">
+            <div className="d-flex flex-column" style={{ width: '400px' }}>
+              <form className="m-0">
+                <div className="form-group">
+                  <h3 className="text-center mt-3">
+                    {intl.formatMessage(messages['logisration.reset.password.page.heading'])}
+                  </h3>
+                  <p className="mb-4">
```

Comment:
Add `mt-4` and decrease the margin bottom to 2. With this the layout looks little odd.

Reply:
I have changed it to resemble Forgot Password page now and now look and feel is more consistent with it.